### ISSUE TITLE
[Feature request] deleted files in stats

### ISSUE THREAD

**Issue by matrix1000** [rclone/rclone#1034]
Hi Nick,

it would be very nice if you can add the information of deletet files in the stat.
Currently (v1.35) you can see in stats: Errors, Checks and Transferred.
It would be perfect if I can also see the deleted files if I am using sync.

Many thanks.

**Comment by ncw** [comment]
Nice idea.

**Comment by chrissv** [comment]
Any plans on this?
Right now I see:
```
Transferred:       99.401M / 742.480 GBytes, 0%, 504.050 kBytes/s, ETA 428h59m41s
Errors:                 0
Checks:             34102 / 34106, 100%
Transferred:           18 / 156654, 0%
Elapsed time:     3m21.9s
```

It would be nice to see:
```
Transferred:       99.401M / 742.480 GBytes, 0%, 504.050 kBytes/s, ETA 428h59m41s
Errors:                 0
Checks:             34102 / 34106, 100%
Transferred:           18 / 156654, 0%
Deleted:                0 / 1024, 0%
Elapsed time:     3m21.9s
```

**Comment by ncw** [comment]
Yes, I agree this would be very nice.  This also came up in #2723 

I'll move it up the priority list.

Any one want to have a go at it?

**Comment by chrissv** [comment]
I code in C, but the GO code has me somewhat baffled. Any pointers where to start, data structures which hold the information?

**Comment by ncw** [comment]
> I code in C, but the GO code has me somewhat baffled. Any pointers where to start, data structures which hold the information?

Looking at the code, I see that the deletes are already being counted, they just aren't displayed to the user.  So it is a question adding `s.deletes` here

https://github.com/ncw/rclone/blob/ea12d76c0354c8c099e253a7701f7dc7d848f95c/fs/accounting/stats.go#L254-L259

I'd quite like not to make another line of stats so I'm not 100% sure how they should be shown.

**Comment by anuar45** [comment]
I am going to use rclone on raspbian for onedrive. Can I pickup this?

**Comment by ncw** [comment]
> I am going to use rclone on raspbian for onedrive. Can I pickup this?

By all means :-)

**Comment by anuar45** [comment]
> 
> 
> > I am going to use rclone on raspbian for onedrive. Can I pickup this?
> 
> By all means :-)

Thanks. 

One question. Why are you not sure about another line in stats?

**Comment by ncw** [comment]
> One question. Why are you not sure about another line in stats?

It just starts making it a bit long.

Maybe the line could appear if there are any deleted files?

Perhaps all the lines in the stats block should only appear if they are non zero?

**Comment by chrissv** [comment]
> Perhaps all the lines in the stats block should only appear if they are non zero?

I really like that idea! Minimalist view unless more info is needed.

**Comment by ncw** [comment]
> > Perhaps all the lines in the stats block should only appear if they are non zero?
> 
> I really like that idea! Minimalist view unless more info is needed.

Give it a try!  We might decide we hate it but I like the idea of being able to add more stats on the fly!

**Comment by anuar45** [comment]
Sorry, I need sometime to get familiar with the codebase.

**Comment by anuar45** [comment]
Sorry, for long waiting...

But adding one fixed field was very straight forward for me. But making this dynamic stats looks very challenging to me and actually I like it.

First of all maybe we can do some stats fixed like(transferred), and other one dynamic(like errors and deletes, checks)?

**Comment by anuar45** [comment]
19 days gone, my progress really blows my minds.

**Comment by ncw** [comment]
> First of all maybe we can do some stats fixed like(transferred), and other one dynamic(like errors and deletes, checks)?

Yes I think some should be fixed and some dynamic most likely.  I'm not sure which yet so I await your proposal :-)

**Comment by anuar45** [comment]
Hi Nick,

Could you please check PR, it is updated?

**Comment by ncw** [comment]
I've merged this to master now (thank you very much @anuar45 ) which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.51
